akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -395,7 +395,8 @@ private[akka] class ShardRegion(
   var handingOff = Set.empty[ActorRef]
   var gracefulShutdownInProgress = false
 
-  import context.dispatcher
+  private implicit val ec: ExecutionContext = context.dispatcher
```

Comment:
@johanandren I think this could be another example where we don't use it correctly in internals. For example `replyToRegionStateQuery` use future operators with this as the implicit EC

Reply:
It would only be dangerous if it is captured inside of a `Future { something()(here) }` block, but it is only referenced in the actor thread where it will always have a value (and as I said somewhere else, the `Future { doSomethingElseAsync()(ctx.implicit <- where it could be nulled out) }` is an anti pattern and when looking through the internals I couldn't find that we do that kind of stuff anywhere.

Reply:
Sorry for being thick here. For some reason I thought it was whenever there was a future callback, but I see now that it must be some flatMap or similar involved, as you already explained in the ticket. Ok then I'm fine without changing the internals.

Regarding samples I think it wouldn't hurt to be more explicit about the EC than the `import`.

For example this would break:
```scala
class Foo(other: ActorRef) extends Actor {

  implicit val timout = Timeout(5.seconds)

  def receive = {
    case msg ⇒
      import context.dispatcher
      val reply = for {
        a ← other ? msg
        b ← other ? a
      } yield b
      reply.pipeTo(sender())
  }

}
```